### PR TITLE
A new gene (and fixing a variable)

### PR DESCRIPTION
**Pull request by The-Moon-Itself**
## About The Pull Request
This pull request adds a new gene to the game: Stable Genetics. It has a instability of 20 and increases your instability limit by 20. While this has no effect on the player if using a mutator, it allows the player to get more mutations if activated. This requirement means that players have to enter the gene scanner to scramble their DNA to ensure they can get the mutation, creating more interaction between players as a geneticist and makes the scannerless consoles for distributing mutations less viable.

This PR also fixes the algorithm used to generate "X"s in the mutation sequence for geneticists to solve, making mutations more accurately resemble their difficulty. While lower difficulty mutations weren't affected much by this algorithm fix, higher difficulty mutations are now noticably harder, and seeking out additional samples with the handheld sequencer or collaborating with your partner would see more play, increasing the depth of the role.

## Why It's Good For The Game
A new mutation to help geneticists reach their full potential designed in a way to increase player interactions.

Difficulty variable on mutations are accurate.

## Testing Photographs and Procedure

<details>

<summary>Difficulty fix</summary>

harder_mutations

</details>

<details>
<summary>Stable genetics</summary>

Stable_Genetics

</details>

## Changelog
:cl:
add: New mutation that lets you get even more mutations.
tweak: Mutations now accurately reflect their difficulty.
/:cl:

### CODE REVIEW

**Comment by The-Moon-Itself** [comment]
bruh, fat fingered the enter key

**Comment by mcmeiler** [comment]
Bro, just fix the tags and reopen it. You don't necessarily need to make a new PR.

**Comment by The-Moon-Itself** [comment]
Not quite sure how to remove tags

**Comment by imsynthex** [comment]
maintainers have to remove tags manually, just don't worry about it p much

File: code/datums/mutations/misc.dm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//mutations that don't fit into the other categories, but aren't complex enough for their own file.
```
Comment (Archanial) [inline code/datums/mutations/misc.dm]: Misc is bad name for this. Just name it stable_genetics

File: code/datums/mutations/stable.dm [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	desc = "The specimen's genome can handle more mutations before melting, reducing instability by 20 if activated."
+	quality = POSITIVE
+	text_gain_indication = "<span class='notice'>You feel healthy.</span>"
+	instability = 20 //Must be activated from your base SE, this isn't completely free stability
+	difficulty = 20 //Good luck solving this without a handheld sequencer
+
+/datum/mutation/human/stable_genetics/on_acquiring(mob/living/carbon/human/H)
+	if(..())
+		return
+	H.dna.base_stability += 20
```
Comment (EvilDragonfiend) [inline code/datums/mutations/stable.dm:13]: these values 20 should be multiplied by `powre_coeff`

File: code/datums/mutations/stable.dm [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+/datum/mutation/human/stable_genetics/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	owner.dna.base_stability -= 20
```
Comment (EvilDragonfiend) [inline code/datums/mutations/stable.dm:18]: the same

**Review by EvilDragonfiend — COMMENTED** [review]
you might want to put a combination with power chromosome. for that, you need to use `powre_coeff` variable and make the primary value a bit adjusted. (or just go as 20.)
to use that value, refer `/datum/mutation/human/firebreath`

**Comment by The-Moon-Itself** [comment]
> you might want to put a combination with power chromosome. for that, you need to use `powre_coeff` variable and make the primary value a bit adjusted. (or just go as 20.) to use that value, refer `/datum/mutation/human/firebreath`

I want to do this, but at the same time I don't want to make the synchronizing chromosome obsolete. either I make the power chromosome really weak, such as only changing the cap increase from 20 to 22, or I make the power chromosome increase instability as well, so synchronizers are still useful for mutators.

**Comment by The-Moon-Itself** [comment]
I've tried to implement a power chromosome onto it, but I was unable to get it to feel right both in terms of balance and mechanics. I think I'll just do without it for now.

**Comment by zeskorion** [comment]
this seems like it has no intent but to allow you to stack more genetics powers. May i point you to the fact that you can install negative genes to lower your instability?

**Comment by The-Moon-Itself** [comment]
> this seems like it has no intent but to allow you to stack more genetics powers. May i point you to the fact that you can install negative genes to lower your instability?

Wait, that works? If so then yeah, this is completely pointless and I’ll close this.

**Comment by The-Moon-Itself** [comment]
Okay. So negative instability does work, but nothing in the game has that, so this PR isn't stepping on anyone's toes. The reason why this mutation has positive instability is that its effects are intended to cancel out when using a mutator instead of an injector, encouraging players to get into the dna scanner and have the geneticist scramble their DNA to get this as a native gene that can be activated.

**Comment by zeskorion** [comment]
> Okay. So negative instability does work, but nothing in the game has that, so this PR isn't stepping on anyone's toes. The reason why this mutation has positive instability is that its effects are intended to cancel out when using a mutator instead of an injector, encouraging players to get into the dna scanner and have the geneticist scramble their DNA to get this as a native gene that can be activated.

this seems like a super convoluted way to add an extra +20 stability, still. I don't think this is the right route to go for giving genetics more gameplay


and- i'm pretty damn sure some neg mutations have neg instability

**Comment by PowerfulBacon** [comment]
I get the feeling this will be confusing in game

**Comment by The-Moon-Itself** [comment]
> I get the feeling this will be confusing in game

yeah, the more I think back on this and how I coded it, the less I like how I implemented it. I'm going to close this. Maybe I'll think of a better way of doing this another time.

**Pull request closed without merging**